cmd/metrics.go

```diff
@@ -89,48 +89,40 @@ func (c *minioCollector) Collect(ch chan<- prometheus.Metric) {
 		return
 	}
 
-	storageAPIs := []StorageAPI{}
-	for _, ep := range globalEndpoints {
-		for _, endpoint := range ep.Endpoints {
-			if endpoint.IsLocal {
-				// Construct storageAPIs.
-				sAPI, _ := newStorageAPI(endpoint)
-				storageAPIs = append(storageAPIs, sAPI)
-			}
-		}
-	}
+	storageInfo := objLayer.StorageInfo(context.Background(), true)
 
-	disksInfo, onlineDisks, offlineDisks := getDisksInfo(storageAPIs)
+	offlineDisks := storageInfo.Backend.OfflineDisks
+	onlineDisks := storageInfo.Backend.OnlineDisks
 	totalDisks := offlineDisks.Merge(onlineDisks)
 
-	for _, offDisks := range offlineDisks {
```

Comment:
What are the implications of this change from a slice v/s single value ?

Reply:
`offlineDisks` is actually a map[`nodeAddress`][`no_of_disks`]. If we filter the local disks alone, there will be only one node Address entry in map. So `offlineDisks.Sum()` will actually give the sum of `no_of_disks` in `offlineDisks` map

Reply:
So it was always a single entry map? Just that this info is filtered at a lower layer.

Reply:
yes it was a single entry map here since we we call `getDisksInfo` with local disks only. It will be a single map with its local peer as the key and offline disks as value